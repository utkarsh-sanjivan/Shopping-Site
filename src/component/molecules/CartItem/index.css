.product-container {
    padding: 10px;
    display: flex;
}

.product-name-container {
    width: 25%;
    display: flex;
    flex-direction: column;
}

.product-image {

}

.product-name {
    font-size: 12px;
    padding-top: 5px;
}

.cart-product-info {
    width: 25%;
    line-height: 70px;
    text-align: center;
    padding: 0 10px;
}

span {
    margin: 0 10px;
    cursor: pointer;
}