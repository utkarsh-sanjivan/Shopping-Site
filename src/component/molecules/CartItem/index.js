import React from 'react';
import './index.css';

export default function CartItem(props) {
    return (
        <div className='cart-product-container'>
            <div className='cart-product-name-container'>
                <img className='cart-product-image' src={props.image} width={50}/>
                <div className='cart-product-name'>{props.name}</div>
            </div>
            <div className='cart-product-info'>₹{props.price}</div>
            <div className='cart-product-info'>
                <span onClick={() => props.removeOne(props.id, props.cartQuantity, props.index)}>-</span>
                {props.cartQuantity}
                <span onClick={() => props.addOne(props.id)}>+</span>
            </div>
            <div className='cart-product-info'>₹{parseInt(props.price)*parseInt(props.cartQuantity)}</div>
        </div>
    );
}