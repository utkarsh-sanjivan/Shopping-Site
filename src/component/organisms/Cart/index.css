.cart-container {
    width: 30%;
    border: 1px solid #dedede;
    padding: 20px 10px;
    margin: 19px 1%;
}

@media only screen and (max-width: 700px) {
    .cart-container {
        width: 50%;
    }
}

@media only screen and (max-width: 400px) {
    .cart-container {
        width: 80%;
        margin: 10px auto;
    }
}

.cart-header {
    display: flex;
    line-height: 40px;
    justify-content: space-between;
}

.header-item {
    width: 25%;
    text-align: center;
}

.cart-footer {
    
}

.footer-item {
    text-align: right;
}