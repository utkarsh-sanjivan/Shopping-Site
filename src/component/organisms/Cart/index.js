import React from 'react';
import './index.css';
import CartItem from '../../molecules/CartItem';

export default function Cart(props) {
    return (
        <div className='cart-container'>
            <div className='cart-header'>
                <div className='header-item'>Product</div>
                <div className='header-item'>Price</div>
                <div className='header-item'>Quantity</div>
                <div className='header-item'>Total</div>
            </div>
            {props.cartItem.map((item, index) => 
                <CartItem 
                    {...item}
                    index={index}
                    key={props.id}
                    removeOne={props.removeOne}
                    addOne={props.addOne}
                />
            )}
            {props.cartItem.length>0?
                <div className='cart-footer'>
                    <hr />
                    <div className='footer-item'>Total: â‚¹{props.totalAmount}</div>
                </div>
            : null}
        </div>
    )
} 