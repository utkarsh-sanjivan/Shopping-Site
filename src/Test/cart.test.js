import React from 'react';
import { create } from "react-test-renderer";
import Cart from '../component/organisms/Cart/index';
import CartItem from '../component/Molecules/CartItem/index';

let wrapper;
let root;
const renderComponent = props => <Cart {...props} />;
const baseProps = {
  cartItem: [],
  addOne: () => {},
  removeOne: () => {},
  totalAmount: () => {},
};

beforeEach(() => {
  wrapper = create(renderComponent(baseProps));
  root=wrapper.root;
});

describe('Testing Cart Component with No Items', () => {
  it('Cart Component should render properly', () => {
    expect(wrapper.toJSON()).toMatchSnapshot();;
  });

  it('should atleast 6 div component', () => {
    expect(root.findAllByType('div').length===6);
  });

  it('should have atleast no Cart Item Component', () => {
    expect(root.findAllByType(CartItem).length===0);
  });
});

describe('Testing Cart Component with Items', () => {
  wrapper = create(renderComponent({ 
    ...baseProps,
    cartItem: [{
      "id": 1,
      "name": "Product 1",
      "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type ",
      "price": "100.00",
      "cartQuantity": "10",
      "image": "http://truth-events.com/wp-content/uploads/2019/09/dummy.jpg"
    },]
  }));
  root=wrapper.root;

  it('Cart Component should render properly', () => {
    expect(wrapper.toJSON()).toMatchSnapshot();;
  });

  it('should atleast 8 div component', () => {
    expect(root.findAllByType('div').length===8);
  });

  it('should have atleast no Cart Item Component', () => {
    expect(root.findAllByType(CartItem).length===0);
  });
});
