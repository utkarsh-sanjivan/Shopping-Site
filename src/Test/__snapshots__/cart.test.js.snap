// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Cart Component with Items Cart Component should render properly 1`] = `
<div
  className="cart-container"
>
  <div
    className="cart-header"
  >
    <div
      className="header-item"
    >
      Product
    </div>
    <div
      className="header-item"
    >
      Price
    </div>
    <div
      className="header-item"
    >
      Quantity
    </div>
    <div
      className="header-item"
    >
      Total
    </div>
  </div>
  <div
    className="cart-product-container"
  >
    <div
      className="cart-product-name-container"
    >
      <img
        alt="Cart Product"
        className="cart-product-image"
        src="http://truth-events.com/wp-content/uploads/2019/09/dummy.jpg"
        width={50}
      />
      <div
        className="cart-product-name"
      >
        Product 1
      </div>
    </div>
    <div
      className="cart-product-info"
    >
      ₹
      100.00
    </div>
    <div
      className="cart-product-info"
    >
      <span
        onClick={[Function]}
      >
        -
      </span>
      10
      <span
        onClick={[Function]}
      >
        +
      </span>
    </div>
    <div
      className="cart-product-info"
    >
      ₹
      1000
    </div>
  </div>
  <div
    className="cart-footer"
  >
    <hr />
    <div
      className="footer-item"
    >
      Total: ₹
    </div>
  </div>
</div>
`;

exports[`Testing Cart Component with No Items Cart Component should render properly 1`] = `
<div
  className="cart-container"
>
  <div
    className="cart-header"
  >
    <div
      className="header-item"
    >
      Product
    </div>
    <div
      className="header-item"
    >
      Price
    </div>
    <div
      className="header-item"
    >
      Quantity
    </div>
    <div
      className="header-item"
    >
      Total
    </div>
  </div>
</div>
`;
